runtime/battery-station/src/parameters.rs

```diff
@@ -365,11 +365,33 @@ parameter_types! {
 }
 
 parameter_type_with_key! {
-    // Well, not every asset is a currency ¯\_(ツ)_/¯
+    // Existential deposits used by orml-tokens.
+    // Only native ZTG and foreign assets should have an existential deposit.
+    // Winning outcome tokens are redeemed completely by the user, losing outcome tokens
+    // are cleaned up automatically. In case of scalar outcomes, the market account can have dust.
+    // Unless LPs use `pool_exit_with_exact_asset_amount`, there can be some dust pool shares remaining.
+    // Explicit match arms are used to ensure new asset types are respected.
     pub ExistentialDeposits: |currency_id: CurrencyId| -> Balance {
         match currency_id {
+            Asset::CategoricalOutcome(_,_) => ExistentialDeposit::get(),
+            Asset::CombinatorialOutcome => ExistentialDeposit::get(),
+            Asset::PoolShare(_)  => ExistentialDeposit::get(),
+            Asset::ScalarOutcome(_,_)  => ExistentialDeposit::get(),
+            #[cfg(feature = "parachain")]
+            Asset::ForeignAsset(id) => {
+                let maybe_metadata = <
+                    orml_asset_registry::Pallet<super::Runtime> as orml_traits::asset_registry::Inspect
+                >::metadata(&Asset::ForeignAsset(*id));
+
+                if let Some(metadata) = maybe_metadata {
+                    return metadata.existential_deposit;
+                }
+
+                1
```

Comment:
The `None` case can only occur if the code queries the `ED` for a `ForeignAsset` that was not registered in the asset registry. Since reserve asset transfers are the only way to mint `ForeignAsset`s and those require that the asset is registered in the metadata, the `None` case highlighted above should never occur in practice.